### PR TITLE
make PR pinging and starting more robust

### PR DESCRIPTION
**Pull request by ericcf**

### CODE REVIEW

File: app/scripts/controllers/usersetup.js [hunk 1]
```diff
@@ -32,26 +32,31 @@
       }
 
       (function() {
-        var pingPurpleRobotPromise = null;
+        var pingStop = null;
+
+        function stopPing() {
+          if (pingStop !== null) {
```
Comment (michaelwehrley) [inline app/scripts/controllers/usersetup.js]: I'm curious: related to what we were talking about before, if you had `pingStop != null` would it fail for `undefined` as well?

File: app/scripts/controllers/usersetup.js [hunk 2]
```diff
@@ -32,26 +32,31 @@
...
+            $interval.cancel(pingStop);
+            pingStiop = null;
```
Comment (michaelwehrley) [inline app/scripts/controllers/usersetup.js]: I think you mean `pingStop`, not `pingStiop`  That would have been a pain to debug.

**Comment by michaelwehrley** [comment]
After you fix `pingStiop`, you can merge. :+1:

File: app/scripts/controllers/usersetup.js (hunk 1 above)
Comment (ericcf) [inline app/scripts/controllers/usersetup.js]: Here, I'm explicitly setting `pingStop` to `null` or to a promise, so it should never be `undefined`.

File: app/scripts/controllers/usersetup.js (hunk 2 above)
Comment (ericcf) [inline app/scripts/controllers/usersetup.js]: Ha, good catch!

**Pull request merged**